### PR TITLE
[Feature] load data on fe support of list partition

### PR DESCRIPTION
**Pull request by hiliuxg**
issue #4035

## What type of PR is this：
- [ ] bug
- [X] feature
- [ ] enhancement
- [ ] others

## main modify on fe side ：
1. Descriptors.thrift add list partition meta values
2. OlapTableSink.java add list partiition metadata to TOlapTablePartition

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 42 / 62 (67.74%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|42|62|67.74%|[316, 317, 318, 327, 328, 329, 330, 335, 336, 337, 341, 351, 352, 353, 358, 359, 360, 381, 382, 383]|

File: fe/fe-core/src/main/java/com/starrocks/planner/OlapTableSink.java [hunk 1]
```diff
@@ -328,6 +312,79 @@ private TOlapTablePartitionParam createPartition(long dbId, OlapTable table) thr
...
+        }
+    }
+
+    private void setDistributedColumns(TOlapTablePartitionParam partitionParam,
+                                       DistributionInfo selectedDistInfo,
+                                       Partition partition,OlapTable table) throws UserException{
+        DistributionInfo distInfo = partition.getDistributionInfo();
+        if (selectedDistInfo == null) {
+            partitionParam.setDistributed_columns(getDistColumns(distInfo, table));
+            selectedDistInfo = distInfo;
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/planner/OlapTableSink.java:379]: Assignment here has no effect on `selectedDistInfo` outside.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 41 / 62 (66.13%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|41|62|66.13%|[316, 317, 318, 327, 328, 329, 330, 335, 336, 337, 341, 351, 352, 353, 358, 359, 360, 381, 382, 383, 386]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 41 / 62 (66.13%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|41|62|66.13%|[316, 317, 318, 327, 328, 329, 330, 335, 336, 337, 341, 351, 352, 353, 358, 359, 360, 381, 382, 383, 386]|

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**